x-pack/legacy/plugins/uptime/server/lib/adapters/monitor_states/elasticsearch_monitor_states_adapter.ts

```diff
@@ -0,0 +1,495 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, set, sortBy } from 'lodash';
+import { DatabaseAdapter } from '../database';
+import { UMMonitorStatesAdapter } from './adapter_types';
+import {
+  MonitorSummary,
+  SummaryHistogram,
+  Check,
+  SnapshotCount,
+} from '../../../../common/graphql/types';
+import { INDEX_NAMES } from '../../../../common/constants';
+import { getHistogramInterval, getFilteredQueryAndStatusFilter } from '../../helper';
+
+type SortChecks = (c: Check) => string[];
+const checksSortBy = (c: Check) => [
+  get<string>(c, 'observer.geo.name'),
+  // TODO: this doesn't _really_ sort by ip, it's a naive string sort
```

Comment:
I don't understand what you mean by this not being real here.

Reply:
I think I see what you mean, we just sort results in memory right? We need the states index to really sort it.

Reply:
We could do that sorting in memory with `searchAfter`, but I think it's not worth the complexity for this PR, and the states index is a better solution.

Reply:
I meant the in-memory sort wouldn't work in all cases, but I'm having trouble producing this, so we can probably drop this comment.

Reply:
Resolved in https://github.com/elastic/kibana/pull/38895/commits/d1ef839d67ad578aec07578eef7b700a1caf2887.